library/s3.py

```diff
@@ -33,10 +33,15 @@ def upload_file(self, name: str, version: str, path: str, acl: str = "public-rea
         response = self.put(path, key, acl)
         return response
 
-    def put(self, path: str, key: str, acl: str = "public-read") -> dict:
+    def put(
+        self, path: str, key: str, acl: str = "public-read", version: str = ""
+    ) -> dict:
```

Comment:
I was thinking maybe we should aim for a more general use case here?
```python
def put(
        self, path: str, key: str, acl: str = "public-read", metadata: dict = {}
    ) -> dict:
```
just in case we might want to extend metadata to include more stuff here. Maybe we can even just throw the entire config in there